### PR TITLE
Feature/components/form

### PR DESCRIPTION
**Pull request by Vomcat**

### CODE REVIEW

**Review by p-szalast — APPROVED** [review]

File: src/components/Cart/Cart.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import { useSelector, useDispatch } from "react-redux";
+
```
Comment (magnowak) [inline src/components/Cart/Cart.tsx:2]: no need for the empty lines here

File: src/components/Cart/CartItem.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import { useDispatch } from "react-redux";
+import styled from "styled-components";
+
```
Comment (magnowak) [inline src/components/Cart/CartItem.tsx:3]: no need for empty lines

File: src/components/Form/Input.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { forwardRef, InputHTMLAttributes } from "react";
+
+import styled from "styled-components";
+
+import { colors, dimensions, respondFrom, breakpoints } from "styles";
+
+interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
+  sizeEl?: "small" | "big";
```
Comment (magnowak) [inline src/components/Form/Input.tsx:8]: is this used anywhere?

File: src/components/Form/Input.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  border: 1px solid ${colors.secondary};
+  padding: ${dimensions.spacing.sm}px ${dimensions.spacing.md}px;
+
+  ${respondFrom(breakpoints.tablet)`
+    padding: ${dimensions.spacing.sm}px ${dimensions.spacing.md}px;
+  `}
+`;
+
+const Input = forwardRef<HTMLInputElement, InputProps>((props, ref) => {
+  const { ...inpputAtributes } = props;
```
Comment (magnowak) [inline src/components/Form/Input.tsx:23]: spelling error

File: src/components/layout/Navigation.tsx [hunk 5]
```diff
@@ -34,16 +46,26 @@ const NavigationLogo = styled.div`
...
+      gap: ${dimensions.spacing.md}px;
+`}
 `;
 
 const Navigation = () => {
+  const [isActiveCart, setIsActiveCart] = useState(false);
+
+  const cartItems = useSelector((state: RootState) => state.items);
+
+  const handleMouseOver = () => {};
```
Comment (magnowak) [inline src/components/layout/Navigation.tsx:67]: is it used?

File: src/pages/Order.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    padding: ${dimensions.spacing.md3}px 0;
+    min-height: calc(100vh - 50px - 82px);
+`}
+`;
+
+const Order = () => {
+  const cartItems = useSelector((state: RootState) => state.items);
+
+  return (
+    <>
```
Comment (magnowak) [inline src/pages/Order.tsx:30]: no need for the empty tag

File: src/interfaces/Cart.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+export interface CartStyleProps {
+  variant: "default" | "menu";
+  items: any[];
+}
+
+export interface CartItemProps {
+  id?: number;
```
Comment (magnowak) [inline src/interfaces/Cart.ts:7]: shouldn't id be obligatory?

File: src/interfaces/Ui.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+export interface Props {
+  children: React.ReactNode;
+}
+
+export interface ButtonProps extends Props {
+  type?: "button" | "submit" | "reset" | undefined;
```
Comment (magnowak) [inline src/interfaces/Ui.ts:6]: do you need to type "undefined" explicitly if you already marked the property as optional?

File: src/store/cartSlice.ts [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+  totalQuantity: 0,
+  error: null,
+};
+
+const cartSlice = createSlice({
+  name: "cart",
+  initialState,
+  reducers: {
+    addItem(state, action) {
+      const newItem = action.payload;
```
Comment (magnowak) [inline src/store/cartSlice.ts:28]: consider object destructuring here.

File: src/components/Cart/Cart.tsx [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+        <h3>Shopping Cart</h3>
+        {items.length > 0 && (
+          <RemoveAllText onClick={removeAllItemsHandler}>
+            Remove All
+          </RemoveAllText>
+        )}
+      </HeaderWrapper>
+      <CartItemWrapper>
+        {items.length > 0
+          ? items.map((item) => (
```
Comment (magnowak) [inline src/components/Cart/Cart.tsx:88]: considering destructuring here, see why: https://github.com/airbnb/javascript#destructuring--object

File: src/components/Product/ProductItem.tsx [hunk 11]
```diff
@@ -40,6 +43,11 @@ const ProductItem: React.FC<ProductItemProps> = ({
   name,
   price,
 }) => {
+  const dispatch = useDispatch();
+
+  const addItemHandler = () => {
+    dispatch(cartActions.addItem({ name, quantity: 1, price, id }));
```
Comment (magnowak) [inline src/components/Product/ProductItem.tsx:49]: for readability I would do: {name, price, id, quantity: 1}

File: src/components/Form/Input.tsx (hunk 4 above)
Comment (Vomcat) [reply src/components/Form/Input.tsx:23]: done

File: src/components/layout/Navigation.tsx (hunk 5 above)
Comment (Vomcat) [reply src/components/layout/Navigation.tsx:67]: Removed

File: src/store/cartSlice.ts (hunk 9 above)
Comment (Vomcat) [reply src/store/cartSlice.ts:28]: Changed

File: src/components/Product/ProductItem.tsx (hunk 11 above)
Comment (Vomcat) [reply src/components/Product/ProductItem.tsx:49]: Changed

File: src/components/Cart/Cart.tsx (hunk 10 above)
Comment (Vomcat) [reply src/components/Cart/Cart.tsx:88]: Changed

File: src/interfaces/Ui.ts (hunk 8 above)
Comment (Vomcat) [reply src/interfaces/Ui.ts:6]: Changed

File: src/interfaces/Cart.ts (hunk 7 above)
Comment (Vomcat) [reply src/interfaces/Cart.ts:7]: Changed

**Pull request merged**